### PR TITLE
feat: dynamic breadcrumbs

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

See issue #3114 for background.

In order to avoid duplicate/different breadcrumb strings, this change uses the existing Route breadcrumb as the name of each page. This requires a one-time change to some of the existing breadcrumbs to match the corresponding action/what we should actually call the page.

We have three special kinds of Routes:
- the main nav lists
- the details pages
- tabs

The first two are the only ones you can go 'back' to, and can only be one (list > action) or two (list > details > action) levels deep. Tabs affect the current URL but don't affect what the 'current' page is. A new navLevel property is added to tag these special routes.

In order to keep things relatively simple we'll just track the first two kinds of routes as properties in a store and set a lastPage whenever we enter a route. Likewise, the current page changes for any route except tabs, but detail routes are special since they need to change the back-link. For instance, if you click on Image Details > Inspect, then Run Image, cancelling should return you to Inspect instead of the image summary.

Someone may ask: why are breadcrumbs put in a store vs a Route property like meta? Because a visual page like Details is rendered from several routes (/, /containers/:id/, /containers/:id/xyz) and getting the main page to have the correct name and URL for a tab when we haven't gotten to that route yet is more complicated than simply sharing a store.

- Detailed change notes -

Route/breadcrumb:
- Use navLevel as described above to keep track of current list/details pages and set last/current page in breadcrumb store.

App:
- Fix breadcrumb names and set basic navLevels.

Details pages:
- Use breadcrumb in DetailsPage, add navLevel to tabs and remove parent from *Details pages.

Form pages:
- Use breadcrumb in FormPage and remove parent from the two pages that use it.
- Added showBreadcrumb property so Help page can still turn it off.
- Set navLevel in RunImage so that the current page name doesn't pick up the tab name.

### Screenshot/screencast of this PR

Screenshot 2023-07-05 at 9 16 08 AM
Screenshot 2023-07-05 at 9 16 31 AM
Screenshot 2023-07-05 at 9 16 42 AM

### What issues does this PR fix or reference?

Fixes issue #3114.

### How to test this PR?

Go to each details page to confirm breadcrumb looks correct. Go to containers list and select a pod to confirm the breadcrumb is now Containers > Pod Details.

Go to each of the form pages: Help (no breadcrumb), Run Image, Deploy Pod to Kubernetes to confirm the page name and breadcrumb at the top. Try these actions from both the list and details pages to confirm the breadcrumb is different and correct in both cases.

On the image details page select the History or Inspect tab before using Run Image, and confirm the breadcrumb or close goes back to the correct tab.

### CODE REVIEW

File: packages/renderer/src/App.svelte [hunk 1]
```diff
@@ -85,59 +85,56 @@ window.events?.receive('display-troubleshooting', () => {
         <Route path="/" breadcrumb="Dashboard Page">
           <DashboardPage />
         </Route>
-        <Route path="/containers" breadcrumb="Containers">
+        <Route path="/containers" breadcrumb="Containers" navLevel="list">
```
Comment (benoitf) [inline packages/renderer/src/App.svelte]: just wondering if we should not then have a `<RouteList>` ,`<RouteTab>`and `<RouteDetails>` elements ?

Comment (benoitf) [reply packages/renderer/src/App.svelte]: or if it's possible to automatically detect that with the `meta` object because we should know what were the routings to go to that page ?

**Review by benoitf — COMMENTED** [review]
Thinking overnight, I come to the fact that we should leverage the convention over the configuration

If the component that we pass ends with `List.svelte` then it has the `navLevel` of `List`

so for example instead of having
```svelte
<Route path="/pods" breadcrumb="Pods" navLevel="list">
   <PodsList />
</Route>
```
we can only use
```svelte
<Route path="/pods" breadcrumb="Pods">
   <PodsList />
</Route>
```

and probably we could also have the breadcrumb being automatically set to the name before List.svelte  so only

```svelte
<Route path="/pods">
   <PodsList />
</Route>
```

The same for details:

```svelte
<Route path="/volumes/:name/:engineId/*" breadcrumb="Volume Details" let:meta navLevel="details">
   <VolumeDetails volumeName="{decodeURI(meta.params.name)}" engineId="{decodeURI(meta.params.engineId)}" />
</Route>
```

If page ends with Details.svelte then it becomes details

for tabs we should also use a convention like `....Tab.svelte`

Advantage is that when looking at a file we know where it is used or the type of component inside it, and breadcrumb/routing can automatically do some default

**Comment by deboer-tim** [comment]
I like that a lot - I think you're right that convention beats configuration for this. There is an unfortunate bit of initial renaming, but that's one-time and straight-forward enough, and if we ever need to make history deeper we could extend the pattern.

In my initial digging I haven't been able to figure out how to get the child component name from the Route though. Using something like the following comes close (besides breaking the layout): https://svelte.dev/repl/7c780140b089441193e55ca509e2bc92?version=3.44.0

**Comment by deboer-tim** [comment]
Another issue - a page like RunImage doesn't use components for its tabs, they're just Route > divs. We might be able to still use convention for the breadcrumb, but since not all Routes have (or should have) breadcrumbs we'd still need something to indicate it.

**Comment by deboer-tim** [comment]
Fixed and added tests.

I've tried a few ways to determine the underlying component name, but I don't see any way to get the pre-rendered content, and cases like RunImage would make a naming convention more complicated. I do see some other conventions we could apply, e.g.:
- tab title = capitalized URL/path
- page name = breadcrumb (e.g. Settings > Registries page title should come from the breadcrumb or vice-versa - the 'current page' in this PR would actually make the first easy)
- default breadcrumb = capitalized path (but some routes are used just to get meta and shouldn't have breadcrumbs, so then we'd need a separate property to disable breadcrumb)
but these are all somewhat independent and the core problem remains that there is nothing else to use that indicates that list, details, and tabs are 'special' routes. 

In other words, I don't see any other short-term way to fix our breadcrumbs other than a change like this, and at least we get one convention applied (details and form page name = breadcrumb name). I'd be happy to investigate any of the others in a separate issue.

**Comment by deboer-tim** [comment]
Rebased to pick up new RunImage tests and added mock breadcrumb.

**Comment by benoitf** [comment]
if we go down that way, probably `navLevel` should be strongly typed with a pre-defined list of available options to avoid errors like 'list' vs 'List' vs not supported attributes

File: packages/renderer/src/Route.svelte [hunk 2]
```diff
@@ -2,12 +2,14 @@
...
 import type { TinroBreadcrumb, TinroRouteMeta } from 'tinro';
 import { TelemetryService } from './TelemetryService';
+import { lastPage, currentPage, listPage, detailsPage } from './stores/breadcrumb';
 
 export let path = '/*';
 export let fallback = false;
 export let redirect = false;
 export let firstmatch = false;
 export let breadcrumb = null;
+export let navLevel = undefined;
```
Comment (benoitf) [inline packages/renderer/src/Route.svelte]: navLevel should be a type like `'list' | 'details' | undefined`...

or an enum like `enum NavLevelType { LIST, DETAILS, UNKNOWN}`

**Review by benoitf — APPROVED** [review]
please add a type for navLevel (instead of any)

File: packages/renderer/src/Route.svelte (hunk 2 above)
Comment (deboer-tim) [reply packages/renderer/src/Route.svelte]: Tried both, and the type felt more natural and concise. New commit pushed with the change, also renamed to NavigationHint.
